lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1463,9 +1463,9 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
     llvm::Metadata *Elements[] = {DBuilder.createMemberType(
         Scope, "", File, 0, SizeInBits, AlignInBits, 0, Flags, UniqueType)};
     return DBuilder.createStructType(
-        Scope, "", File, Line, SizeInBits, AlignInBits, Flags,
+        Scope, Name, File, Line, SizeInBits, AlignInBits, Flags,
         /* DerivedFrom */ nullptr, DBuilder.getOrCreateArray(Elements),
-        llvm::dwarf::DW_LANG_Swift, nullptr, "", SpecificationOf, 0);
+        llvm::dwarf::DW_LANG_Swift, nullptr, MangledName, SpecificationOf, 0);
```

Comment:
Careful — this might have been deliberate. We need to ensure the only types that have a mangled name are the ones were it is guaranteed that all copies of all types with the same mangled name will be identical in all CUs. LTO type uniquing works by merging all types with the same LinkageName.